### ISSUE TITLE
Unkown commands in renderQueue

### ISSUE THREAD

**Issue by minggo** [cocos2d/cocos2d-x#8534]
# Note

 This issue is migrated from [here](http://www.cocos2d-x.org/issues/5877). It was created at 2014/09/01 02:25:32 +0000
# Description

How to reproduce this issue：
1. use  `cocos new -l cpp` to create a new project
2. add the following code in the HelloWorldScene's init method:

```
    auto renderTexture = RenderTexture::create(visibleSize.width, visibleSize.height);
    renderTexture->setPosition(Vec2(visibleSize.width/2, visibleSize.height/2));
    this->addChild(renderTexture, 10);

    auto bgSprite = Sprite::create("HelloWorld.png");
    bgSprite->setAnchorPoint(Vec2(0.5,0.5));
    bgSprite->setPosition(center);
    renderTexture->begin();
    bgSprite->visit();
    renderTexture->end();
```
1. build and run the project, you will see the console output this error message.
2. wushi333333 added comment:
   我也遇到这个问题了，如果解决了，希望能通知我,我用的是3.1.1，wushi333333@163.com.谢谢
- dabingnn added comment:
  There is  an thread opened in the forum, which is used for talking about the bug.
  Link:
  http://discuss.cocos2d-x.org/t/bugs-for-unknown-commands-in-renderer/16865

We can discuss it in detail there
Thanks.
- dabingnn added comment:
  As discussed in the forum, the bug could be solved like this:
- Basically, renderCommand could only be added to renderer in the function `Director::DrawScene()`
- If there is any command added to render out of the `Director::DrawScene()`, user should called `renderer->render()` manually.

This is not a final solution, we will continue working on this to find a better solution.
This is is going to be postponed

**Comment by minggo** [comment]
I tested it with latest codes, and could not reproduce it, so close it.
